File: JavaScript/프로그래머스/2/388352. 비밀 코드 해독/비밀 코드 해독.js
function solution(n, q, ans) {
    function getCombinations(arr, selectNum) {
        if (selectNum === 0) return [[]];
        if (arr.length === 0) return [];
        
        const [first, ...rest] = arr;
        const withFirst = getCombinations(rest, selectNum - 1).map(comb => [first, ...comb]);
        const withoutFirst = getCombinations(rest, selectNum);
        
        return [...withFirst, ...withoutFirst];
    }

    // 1부터 n까지의 수에서 5개를 선택하는 모든 조합 생성
    const allCombinations = getCombinations([...Array(n).keys()].map(x => x + 1), 5);
    let validCount = 0;

    for (const combination of allCombinations) {
        let isValid = true;

        for (let i = 0; i < q.length; i++) {
            const matchCount = q[i].filter(num => combination.includes(num)).length;
            if (matchCount !== ans[i]) {
                isValid = false;
                break;
            }
        }

        if (isValid) validCount++;
    }

    return validCount;
}

File: Python3/프로그래머스/2/388352. 비밀 코드 해독/비밀 코드 해독.py
from itertools import combinations


def solution(n, q, ans):
    possible_codes = list(combinations(range(1, n+1), 5))  # 1~n에서 5개 뽑기
    valid_count = 0

    for code in possible_codes:
        valid = True
        for i in range(len(q)):
            if sum(1 for num in q[i] if num in code) != ans[i]:  # 조건 검사
                valid = False
                break
        if valid:
            valid_count += 1  # 조건을 만족하는 코드 개수 증가

    return valid_count
